### PR TITLE
Make xml parsing more consistent and add tests

### PR DESCRIPTION
**Pull request by daschult**
Resolves https://github.com/Azure/ms-rest-js/issues/245.

### CODE REVIEW

File: test/shared/policies/deserializationPolicyTests.ts [hunk 1]
```diff
@@ -62,7 +62,7 @@ describe("deserializationPolicy", () => {
     assert.deepStrictEqual(response.parsedBody, [123, 456, 789]);
   });
 
-  it("should parse a JSON response body with an uppercase Content-Type", async function() {
+  it("should parse a JSON response body with an uppercase Content-Type", async () => {
```
Comment (amarzavery) [inline test/shared/policies/deserializationPolicyTests.ts]: arrow functions do not work well in mocha https://glebbahmutov.com/blog/test-closures/, https://github.com/mochajs/mocha/issues/2018

File: lib/policies/deserializationPolicy.ts [hunk 2]
```diff
@@ -11,29 +11,56 @@ import { parseXML } from "../util/xml";
...
+export interface DeserializationContentTypes {
+  /**
+   * The content-types that indicate that an operation response should be deserialized as XML.
+   * Defaults to [ "application/json", "text/json" ].
+   */
+  xml?: string[];
+
+  /**
+   * The content-types that indicate that an operation response should be deserialized as JSON.
+   * Defaults to [ "application/xml", "application/atom+xml" ].
```
Comment (kpajdzik) [inline lib/policies/deserializationPolicy.ts]: Aren't defaults switched?

File: test/shared/policies/deserializationPolicyTests.ts (hunk 1 above)
Comment (daschult) [reply test/shared/policies/deserializationPolicyTests.ts]: Changed it back.

File: lib/policies/deserializationPolicy.ts (hunk 2 above)
Comment (daschult) [reply lib/policies/deserializationPolicy.ts]: Yes, they are. I'll fix that.

**Pull request merged**